spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/src/main/java/org/springframework/cloud/dataflow/module/deployer/kubernetes/KubernetesModuleDeployer.java

```diff
@@ -0,0 +1,202 @@
+		}
+		catch (KubernetesClientException e) {
+			logger.warn(e.getMessage(), e);
+			return buildModuleStatus(id, null);
+		}
+	}
+	
+	@Override
+	public Map<ModuleDeploymentId, ModuleStatus> status() {
+		Map<ModuleDeploymentId, ModuleStatus> result = new HashMap<>();
+		
+		ReplicationControllerList list = 
+				kubernetesClient.replicationControllers()
+				.withLabel(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).list();
+		
+		for (ReplicationController rc : list.getItems()) {
+			Map<String, String> labels = rc.getMetadata().getLabels();			
+			String group = labels.get(SCSM_GROUP_KEY);
+			String label = labels.get(SCSM_LABEL_KEY);			
+			ModuleDeploymentId id = new ModuleDeploymentId(group, label);
+			
+			PodList pods = kubernetesClient.pods().withLabels(labels).list();			
+			result.put(id, buildModuleStatus(id, pods));	
+		}		
+		return result;		
+	}
+
+	private ReplicationController createReplicationController(ModuleDeploymentId id, ModuleDeploymentRequest request, int externalPort) {
+        ReplicationController rc = new ReplicationControllerBuilder()
+            .withNewMetadata()
+            	.withName(createKubernetesName(id))  // does not allow . in the name
+            	.withLabels(createIdMap(id))
+            	.addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE)
+        		.addToLabels(SCSM_ARTIFACT_ID, request.getCoordinates().getArtifactId())
+        		.addToLabels(SCSM_GROUP_ID, request.getCoordinates().getGroupId())
+        		.addToLabels(SCSM_VERSION, request.getCoordinates().getVersion())
+        		.addToLabels(SCSM_EXTENSION, request.getCoordinates().getExtension())
+            .endMetadata()
+            .withNewSpec()
+            	.withReplicas(request.getCount())
+            	.withSelector(createIdMap(id))
+            .withNewTemplate()
+            	.withNewMetadata()
+            		.withLabels(createIdMap(id))
+                	.addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE)
+            	.endMetadata()
+                .withSpec(createPodSpec(request, externalPort))	                	
+            .endTemplate()
+            .endSpec().build();        
+       
+        return kubernetesClient.replicationControllers().create(rc);        
+	}
+	
+	private PodSpec createPodSpec(ModuleDeploymentRequest request, int port) {
+		PodSpecBuilder podSpec = new PodSpecBuilder();
+
+		// Add image secrets if set
+		if (properties.getImagePullSecret() != null) {
+			podSpec.addNewImagePullSecret(properties.getImagePullSecret());
+		}
+		podSpec.addToContainers(containerFactory.create(request, port));
+		return podSpec.build();
+	}
+
+	private void createService(ModuleDeploymentId id, ModuleDeploymentRequest request, int externalPort) {
+		kubernetesClient.services().inNamespace(kubernetesClient.getNamespace()).createNew()
+			.withNewMetadata()
+              .withName(KubernetesUtils.createKubernetesName(id)) // does not allow . in the name
+              .withLabels(createIdMap(id))
```

Comment:
Bug in my code. it should only be label based not name based. The issue is that the name of a `service` in Kubernetes falls under certain restrictions (see my other issue) so a name needs to be deduced that may not reassemble the `group` and `name`/`label`.